src/libraries/System.Formats.Tar/tests/TarEntry/TarEntryGnu.Tests.cs

```diff
@@ -38,6 +38,220 @@ public void Constructor_UnsupportedEntryTypes()
             Assert.Throws<InvalidOperationException>(() => new GnuTarEntry(TarEntryType.LongPath, InitialEntryName));
         }
 
+        [Fact]
+        public void Constructor_ConversionFromV7()
+        {
+            V7TarEntry v7 = new V7TarEntry(TarEntryType.V7RegularFile, InitialEntryName);
+            GnuTarEntry convertedV7 = new GnuTarEntry(other: v7);
+
+            Assert.Equal(TarEntryType.RegularFile, convertedV7.EntryType);
+            Assert.Equal(InitialEntryName, convertedV7.Name);
+        }
+
+        [Fact]
+        public void Constructor_ConversionFromUstar()
+        {
+            UstarTarEntry ustar = new UstarTarEntry(TarEntryType.RegularFile, InitialEntryName);
+            GnuTarEntry convertedUstar = new GnuTarEntry(other: ustar);
+
+            Assert.Equal(TarEntryType.RegularFile, convertedUstar.EntryType);
+            Assert.Equal(InitialEntryName, convertedUstar.Name);
+        }
+
+        [Fact]
+        public void Constructor_ConversionFromPax()
+        {
+            PaxTarEntry pax = new PaxTarEntry(TarEntryType.RegularFile, InitialEntryName);
+            GnuTarEntry convertedPax = new GnuTarEntry(other: pax);
+
+            Assert.Equal(TarEntryType.RegularFile, convertedPax.EntryType);
+            Assert.Equal(InitialEntryName, convertedPax.Name);
+        }
+
+        [Fact]
+        public void Constructor_ConversionFromV7_From_UnseekableTarReader()
+        {
+            using MemoryStream source = GetTarMemoryStream(CompressionMethod.Uncompressed, TestTarFormat.v7, "file");
+            using WrappedStream wrappedSource = new WrappedStream(source, canRead: true, canWrite: false, canSeek: false);
+
+            using TarReader sourceReader = new TarReader(wrappedSource, leaveOpen: true);
+            V7TarEntry v7Entry = sourceReader.GetNextEntry(copyData: false) as V7TarEntry;
+            GnuTarEntry gnuEntry = new GnuTarEntry(other: v7Entry); // Convert, and avoid advancing wrappedSource position
+
+            using MemoryStream destination = new MemoryStream();
+            using (TarWriter writer = new TarWriter(destination, TarEntryFormat.Gnu, leaveOpen: true))
```

Comment:
In these tests it would be a stronger statement if `TarWriter.Format` wasn't equal to the format you were writing... so it showed that even though it was going to use Ustar for WriteEntry(string,string) that it used the provided entry type when writing with WriteEntry(TarEntry)

Reply:
I need to add separate tests for the _default_ format when writing. This test's purpose isn't for checking that the default format for the TarWriter works.

Reply:
> This test's purpose isn't for checking that the default format for the TarWriter works.

Right.  But if the test said the default format was anything other than GNU (for this specific instance of it), then the test would show that WriteEntry(TarEntry) is independent of the default format.

Reply:
Addressed in https://github.com/dotnet/runtime/pull/70325/commits/6fbb257bd25a14c68ff9ec804c241fffff1635f8

Reply:
I don't see how this is resolved in that commit.  It doesn't seem to be creating TarWriters using one value as the default format, then writing entries using some other format to show that WriteEntry(TarEntry) doesn't convert the input.

I still see things using the original pattern at places like https://github.com/dotnet/runtime/pull/70325/files#diff-56194df236171cb8d55a18fe1ef116fd9a2da16fbdcfca6e156358e6a494abc4R75-R78

Reply:
Ok I see what you mean. I cannot _remove_ the parameter yet, because in this PR I have not removed the TarWriter constructor that takes an `IEnumerable<KVP<string, string>>`, so if I omit the format argument, the build fails due to an ambiguity between the two constructors.

What I can do, is make this tests Theories and make them take a format argument, make sure the TarWriter works with all the formats.

Reply:
Fixed in this commit: https://github.com/dotnet/runtime/pull/70325/commits/f23d24427147daab4bd32362a9cb8b975bb1917c